sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -560,7 +562,8 @@ abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
             // [COUNT(DISTINCT bar), COUNT(DISTINCT foo)] is disallowed because those two distinct
             // aggregates have different column expressions.
             val distinctExpressions =
-              functionsWithDistinct.head.aggregateFunction.children.filterNot(_.foldable)
```

Comment:
Sorry, I'm a little confused. Why do we change here? Since all children are semantically equivalent, we can just pick the first distinct function. If we need to look up the child later, we should make sure it uses `ExpressionSet`.

Reply:
>Sorry, I'm a little confused.

Ah yes, possibly I was too. I had not read all of `planAggregateWithOneDistinct` yet, and I see the creation of `rewrittenDistinctFunctions`, where I can possibly take advantage of semantic equivalence.